web/source/keymanweb.js

```diff
@@ -1360,106 +1360,139 @@ if(!window['tavultesoft']['keymanweb']['initialized']) {
      * @param {string|Object} x keyboard name string or keyboard metadata JSON object
      * 
      */  
-    keymanweb.addKeyboardArray = function(x)
-    {
+    keymanweb.addKeyboardArray = function(x) {
       // Store all keyboard meta-data for registering later if called before initialization
-      if(!keymanweb['initialized']) 
-      {
-        for(var k=0; k<x.length; k++)
-          keymanweb.deferredStubs.push(x[k]); 
-        return; 
+      if(!keymanweb['initialized']) {
+        for(var k=0; k<x.length; k++) {
+          keymanweb.deferredStubs.push(x[k]);
+        }
+        return;
       }
   
       // Ignore empty array passed as argument
-      if(x.length == 0) return;
+      if(x.length == 0) {
+        return;
+      }
     
       // Create a temporary array of metadata objects from the arguments used
-      var i,j,kp,kbid,lgid,kvid,cmd='',comma='';      
-      keymanweb.cloudList=[];  
-      for(i=0; i<x.length; i++)
-      {
-        if(typeof(x[i]) == 'string' && x[i].length > 0) 
-        {  
+      var i,j,kp,kbid,lgid,kvid,cmd='',comma='';
+      keymanweb.cloudList=[];
+
+      /**
+       * Function       isUniqueRequest
+       * Scope          Private
+       * @param         tEntry      {object} 
+       * Description    Checks to ensure that the stub isn't already loaded within KMW or subject
+       *                to an already-pending request.
+       */
+      var isUniqueRequest = function(tEntry) {
+        var k;
+
+        if(keymanweb.findStub(tEntry['id'], tEntry['language']) == null) {
+          for(k=0; k < keymanweb.cloudList.length; k++) {
+            if(keymanweb.cloudList[k]['id'] == tEntry['id'] && keymanweb.cloudList[k]['language'] == tEntry['language']) {
+              return false;
+            }
+          }
+          return true;
+        } else {
+          return false;
+        }
+      };
+
+      for(i=0; i<x.length; i++) {
+        if(typeof(x[i]) == 'string' && x[i].length > 0) {
           var pList=x[i].split('@'),lList=[''],tEntry;
-          if(pList[0].toLowerCase() == 'english') pList[0] = 'us';
-          if(pList.length > 1)
-          {
+          if(pList[0].toLowerCase() == 'english') {
+            pList[0] = 'us';
```

Comment:
Is this test appropriate now?

Reply:
I would imagine so, though I suppose I'm not fully sure what this test initially addressed.  My perception was that it corrects an attempt to load "id:english, lang:english" to "id:us, lang:english" for better compatibility with the keyboard cloud API?  Perhaps the default english keyboard was renamed at some point in Tavultesoft development?

Reply:
I am guessing it's something we can remove in the future. Not sure exactly what it was, but hardcoded references like this give me the heebie jeebies. Doesn't need to change in this PR.